sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -2585,9 +2586,12 @@ class Dataset[T] private[sql](
    * Creates a local temporary view using the given name. The lifetime of this
    * temporary view is tied to the [[SparkSession]] that was used to create this Dataset.
    *
+   * @throws AnalysisException if the view name is invalid
+   *
    * @group basic
    * @since 2.0.0
    */
+  @throws[AnalysisException]
```

Comment:
@hvanhovell . Before this PR, this exception is missed. When we add this, there are side effects on Java code.